leakpro/user_inputs/cifar10_gia_input_handler.py

```diff
@@ -0,0 +1,82 @@
+"""Module containing the class to handle the user input for the CIFAR10 dataset."""
+
+import logging
+from collections import OrderedDict
+from copy import deepcopy
+
+import torch
+from torch import cuda, device
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from leakpro.fl_utils.gia_optimizers import MetaAdam, MetaOptimizer, MetaSGD
+from leakpro.fl_utils.gia_train import MetaModule
+from leakpro.import_helper import Self
+from leakpro.user_inputs.abstract_gia_input_handler import AbstractGIAInputHandler
+
+
+class Cifar10GIAInputHandler(AbstractGIAInputHandler):
+    """Class to handle the user input for the CIFAR10 dataset."""
+
+    def __init__(self:Self, configs: dict, logger:logging.Logger, client_data: DataLoader, target_model: torch.nn.Module) -> None:
+        super().__init__(configs, logger, client_data, target_model)
+        self.criterion = self.get_criterion()
+
+    def get_criterion(self:Self)->None:
+        """Set the CrossEntropyLoss for the model."""
+        return torch.nn.CrossEntropyLoss()
+
+    def get_optimizer(self: Self) -> MetaOptimizer:
+        """Set the optimizer for the model."""
+        optimizer = self.configs["audit"]["gia_settings"]["optimizer"]
+        lr = self.configs["audit"]["gia_settings"]["learning_rate"]
+        if optimizer == "SGD":
```

Comment:
This mapping should be in the abstract class, the user should not have to know about this.